### DISCUSSION TITLE
Which syntax definition formats are supported?

### DISCUSSION THREAD

**Discussion by nk9** [github-linguist/linguist#6222] [General]
I am trying to figure out which syntax definition formats and versions are supported by Linguist. I was expecting a list in the docs, but I haven't been able to find one.

Ultimately, I'm hoping to use a `.sublime-syntax` file (YAML), either version 1 or 2. I think I see examples of this format in use already, but I wanted to check. (<s>Ex: the Solidity format [is working](https://github.com/ethereum/consensus-specs/blob/dev/solidity_deposit_contract/deposit_contract.sol) but is only defined [using `.sublime-syntax`](https://github.com/davidhq/SublimeEthereum).</s> I see that they have a `linguist` branch which has a `.tmLanguage` file specifically for Linguist.)

**Comment by lildude** [comment]
The grammar compiler looks for these extensions and expects the format associated with them:

https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/tools/grammars/compiler/loader_fs.go#L18-L24

The compiler doesn't support the newer `.sublime-syntax` grammar format.

**Reply by ratijas** [reply]
Hi, I'm looking to update QML syntax highlighting used by linguist. But I'm having a hard time trying to understand what this tool expects. From what I see, all those plist, tmLanguage, yaml, json and cson (never even heard of it, wow) are merely encoding formats for the same underlying data structures, basically for tmLanguage from TextMate times, is that correct?

My updated plugin is based on newer Sublime Text engine, and written in *.sublime-syntax YAML. I wonder what are my options to get it into Linguist? My plugin uses couple of modern features, like lookahead branching and `pop: 2` stack management patterns, so I doubt it is even possible to downgrade it back to tmLanguage without loss of features.

**Reply by lildude** [reply]
>  are merely encoding formats for the same underlying data structures, basically for tmLanguage from TextMate times, is that correct?

Essentially, yes.

> My updated plugin is based on newer Sublime Text engine, and written in *.sublime-syntax YAML. I wonder what are my options to get it into Linguist? 

Your only option would be to also include a format Linguist does support in your repo. You'll see several other grammars include a Sublime _and_ a VSCode/TextMate-compatible grammar files, often generated off the same source. https://github.com/ziglang/sublime-zig-language is one example that quickly comes to mind.

**Reply by ratijas** [reply]
IIUC zig generates rich *.sublime-syntax file from a simpler .YAML-tmLanguage format. Which isn't going to magically bring you the new exciting features, so I wonder why do they even bother…
